fixtures/flight-vite/server/region.js

```diff
@@ -0,0 +1,214 @@
+'use strict';
+
+// This is a server to host data-local resources like databases and RSC
+
+const path = require('path');
+const url = require('url');
+
+if (typeof fetch === 'undefined') {
+  // Patch fetch for earlier Node versions.
+  global.fetch = require('undici').fetch;
+}
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const busboy = require('busboy');
+const compress = require('compression');
+const {Readable} = require('node:stream');
+const {readFile} = require('fs').promises;
+const React = require('react');
+
+async function createApp() {
+  // Application
+  const app = express();
+  app.use(compress());
+
+  const moduleBasePath = new URL('../src', url.pathToFileURL(__filename)).href;
+
+  let loadModule;
+  if (process.env.NODE_ENV === 'development') {
+    const vite = await import('vite');
+    const {default: reactRefresh} = await import('@vitejs/plugin-react');
+    const {default: reactServer} = await import('react-server-dom-vite/plugin');
+    const viteServer = await vite.createServer({
+      appType: 'custom',
+      server: {middlewareMode: true, hmr: {port: 9898}},
+      resolve: {
+        conditions: ['node', 'import', 'react-server', 'production'],
```

Comment:
Should this have the `'production'` condition when running in `NODE_ENV === 'development'`?